File: 392-is-subsequence/is-subsequence.py
class Solution(object):
    def isSubsequence(self, s, t):
        
        p1 = 0
        p2 = 0
        count = len(s)

        while p1 < len(s) and p2 < len(t):

            if s[p1] == t[p2]:
                count -= 1
                p1 += 1
            p2 += 1
        
        return count == 0

        

File: 392-is-subsequence/is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {

        int p1 = 0, p2 = 0;
        int count = s.length();
        
        while (p1 < s.length() && p2< t.length()) {
            if (s.charAt(p1) == t.charAt(p2)){
                count--;
                p1++;
            }
            p2++;
        }
        return count == 0;
    }
}